File: Codes/1922-count-good-numbers.java
/*
 * @Author: LetMeFly
 * @Date: 2025-04-13 17:07:43
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-04-14 00:07:21
 * @Description: AC,100.00%,87.23%
 */
class Solution {
    private final long mod = 1000000007;

    private long pow(long a, long b) {
        long ans = 1;
        while (b > 0) {
            if ((b & 1) == 1) {
                ans = ans * a % mod;
            }
            a = a * a % mod;
            b >>= 1;
        }
        return ans;
    }

    public int countGoodNumbers(long n) {
        return (int)(pow(5, (n + 1) / 2) * pow(4, n / 2) % mod);
    }
}

File: Codes/1922-count-good-numbers.cpp
/*
 * @Author: LetMeFly
 * @Date: 2025-04-13 17:00:20
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-04-13 17:06:07
 * @Description: AC,100.00%,51.97%
 */
#if defined(_WIN32) || defined(__APPLE__)
#include "_[1,2]toVector.h"
#endif

typedef long long ll;
const ll MOD = 1e9 + 7;

class Solution {
private:
    ll pow(ll a, ll b) {
        ll ans = 1;
        while (b) {
            if (b & 1) {
                ans = ans * a % MOD;
            }
            a = a * a % MOD;
            b >>= 1;
        }
        return ans;
    }
public:
    int countGoodNumbers(long long n) {
        return pow(5, (n + 1) / 2) * pow(4, n / 2) % MOD;
    }
};